### DISCUSSION TITLE
Help with rule for pool pump, chlorine generator and current sensor.

### DISCUSSION THREAD

**Discussion by snowzach** [arendst/Tasmota#15762] [Q&A] [answered]
Hi, I have a Sonoff Dual R3 hooked up to my pool filter and chlorine generator. The chlorine generator has it's own timer that you can't really override. The pool filter has to be running in order for the Chlorine generator to work. What I would like to do ideally is to detect when Chlorine generator kicks and then turn on the pump for say 4 hours. 

The chlorine generator runs about 1A/100W on the second switch. Is there any way to trigger the first switch on for 4 hours when the current sensor jumps up?

```
04:31:08.844 MQT: tele/pool-control/STATE = {"Time":"2022-06-06T04:31:08","Uptime":"1T04:50:10","UptimeSec":103810,"Heap":148,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":22,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":40},"POWER1":"ON","POWER2":"ON","Wifi":{"AP":1,"SSId":"Internal","BSSId":"6A:D7:9A:2A:9C:80","Channel":11,"Mode":"11n","RSSI":50,"Signal":-75,"LinkCount":1,"Downtime":"0T00:00:04"}}
04:31:08.864 MQT: tele/pool-control/SENSOR = {"Time":"2022-06-06T04:31:08","Switch1":"ON","Switch2":"ON","ENERGY":{"TotalStartTime":"2022-05-24T04:00:04","Total":7.534,"Yesterday":6.273,"Today":1.180,"Period":[21, 1],"Power":[256, 7],"ApparentPower":[258,14],"ReactivePower":[38,12],"Factor":[0.99,0.48],"Voltage":127,"Current":[2.043,0.111],"BL09XX":{"Temperature":39.4}},"ESP32":{"Temperature":36.1},"TempUnit":"C"}
```

I would love it if there was an easier way to detect these things and understand what was available from the triggers. 
Thanks!

**Comment by barbudor** [comment]
Unfortunately, the power measurment is global to the R3, so you can't distinguish the power consumption of the chlorine generator from the consumption of the pump itself.
The safest solution would be to use 2 separate devices.

Possible solution: assuming that the pump is always off by default and the chlorine always starts 1st
```
Pulsetime1 14500
Rule1 on Energy#Power>80 do power1 on endon
Rule1 1
Rule1 5
```
This will trigger the pump for 4 hours as soon as the power consumption is above 80W
However, this won't make a difference if the pump is started manually from the web button or MQTT
Using the "Once" mode (Rule1 5) insure that the Pulsetime won't be re-triggered continuously. So total power consumption (whether it generator or pump) has to drop below 80W to re-arm and allow a new trigger.

If the pump and the generator have 2 different well defined and stable power consumption, it may be possible to improve knowing the exact power consumption for all combinations.

**Reply by sfromis** [reply]
With [SetOption129](https://tasmota.github.io/docs/Commands/#setoption129), you can enable split energy consumption data for the Sonoff Dual R3, that's what I'm doing. And then a rule like this can work:
`on ENERGY#Power[2]>0 do .... endon`

**Reply by barbudor** [reply]
Oh my bad
I didn't remember this one has separated power measurement 🤦

**Comment by snowzach** [comment, marked as answer]
That's perfect! Thanks so much!!!

**Reply by snowzach** [reply]
The final config I used was. It works perfectly... 
```
SetOption129 1    # Split out energy results per relay
Pulsetime1 18000  # Turn power1 back off after 5 hours
Rule1 on ENERGY#Power[2]>50 do power1 on endon # When Power[2] is over 50 watts, turn on power 1
Rule1 1           # Enable rule 1
Rule1 5           # Rule 1 one shot mode
```

**Comment by snowzach** [comment]
I ended up updating this again... There's a shock cycle for the chlorine generator which lasts longer than 5 hours... So I have it trigger the pump on when the wattage goes above 50W on the chlorine generator an another rule that triggers 2 hours after it shuts off to turn the pump off.

```
Rule1 ON ENERGY#Power[2]>50 DO power1 ON ENDON # When Power[2] is over 50 watts, turn on power 1
Rule1 1           # Enable rule
Rule1 5           # Rule one shot mode
# When power[2]<50W trigger timer for 2 hours. After timer turn off power 1
Rule2 ON ENERGY#Power[2]<50 DO RuleTimer1 7200 ENDON ON Rules#Timer=1 DO Power1 off ENDON 
Rule2 1           # Enable rule
Rule2 5           # Rule one shot mode
```
